File: examples/onchain-job-queue/web2/springboot/src/main/java/com/example/demo/JobQueueController.java
package com.example.demo;

import org.springframework.web.bind.annotation.*;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import java.util.*;
import java.util.concurrent.ConcurrentHashMap;
import java.util.concurrent.atomic.AtomicInteger;

@RestController
@RequestMapping("/api/jobs")
public class JobQueueController {

    public static class Job {
        public int id;
        public String task;
        public String status = "PENDING";
        public double bounty;
        public String worker;
    }

    private final Map<Integer, Job> jobs = new ConcurrentHashMap<>();
    private final AtomicInteger counter = new AtomicInteger();

    @PostMapping
    public ResponseEntity<?> createJob(@RequestBody Map<String, Object> req) {
        Job job = new Job();
        job.id = counter.incrementAndGet();
        job.task = (String) req.get("task");
        job.bounty = Double.parseDouble(req.get("bounty").toString());
        jobs.put(job.id, job);
        return new ResponseEntity<>(job, HttpStatus.CREATED);
    }

    @PostMapping("/{id}/claim")
    public ResponseEntity<?> claimJob(@PathVariable int id, @RequestBody Map<String, String> req) {
        Job job = jobs.get(id);
        if (job == null || !job.status.equals("PENDING")) {
            return ResponseEntity.badRequest().body("Job not available");
        }
        job.status = "CLAIMED";
        job.worker = req.get("worker");
        return ResponseEntity.ok(job);
    }
}

File: examples/onchain-job-queue/web2/csharp/job-queue-api/Controllers/JobQueueController.cs
using Microsoft.AspNetCore.Mvc;
using System.Collections.Concurrent;

namespace job_queue_api.Controllers;

[ApiController]
[Route("api/jobs")]
public class JobQueueController : ControllerBase
{
    public class Job { public int Id { get; set; } public string Task { get; set; } = ""; public string Status { get; set; } = "Pending"; public double Bounty { get; set; } public string? Worker { get; set; } }
    
    private static readonly ConcurrentDictionary<int, Job> _jobs = new();
    private static int _counter = 0;

    [HttpPost]
    public IActionResult Create([FromBody] dynamic req)
    {
        var id = Interlocked.Increment(ref _counter);
        var job = new Job { Id = id, Task = req.task, Bounty = req.bounty };
        _jobs[id] = job;
        return Created("", job);
    }

    [HttpPost("{id}/claim")]
    public IActionResult Claim(int id, [FromBody] dynamic req)
    {
        if (!_jobs.TryGetValue(id, out var job) || job.Status != "Pending")
            return BadRequest("Job unavailable");

        job.Status = "Claimed";
        job.Worker = req.worker;
        return Ok(job);
    }
}
